File: verification/verify_live_game.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  try {
      console.log('Navigating...');
      await page.goto('http://localhost:8000');
      await page.waitForTimeout(1000);

      // New League
      console.log('Starting league...');
      await page.evaluate(async () => {
          if (window.gameController) await window.gameController.startNewLeague();
      });
      await page.waitForSelector('#onboardModal');
      await page.click('#onboardStart');
      await page.waitForSelector('#hub', { state: 'visible', timeout: 20000 });

      // Live Game
      console.log('Starting live game...');
      await page.evaluate(() => {
          const userTeam = window.state.userTeamId;
          const oppTeam = userTeam === 0 ? 1 : 0;
          window.watchLiveGame(userTeam, oppTeam);
      });
      await page.waitForSelector('#game-sim', { state: 'visible' });
      await page.waitForTimeout(2000); // Wait for animations

      // Take Screenshot
      console.log('Taking screenshot...');
      await page.screenshot({ path: 'verification/live_game_ui.png' });
      console.log('Screenshot saved.');

  } catch (e) {
      console.error(e);
  } finally {
      await browser.close();
  }
})();

File: verification/verify_live_game.py
from playwright.sync_api import sync_playwright

def verify_live_game(page):
    page.on("console", lambda msg: print(f"PAGE LOG: {msg.text}"))
    page.on("pageerror", lambda exc: print(f"PAGE ERROR: {exc}"))

    # 1. Navigate to app
    page.goto("http://localhost:3000")

    # 2. Start New League
    page.get_by_role("button", name="New League").click()

    # 3. Wait for dashboard
    # Increase timeout
    try:
        page.wait_for_selector("text=Season s1", timeout=15000)
    except Exception as e:
        print("Timeout waiting for Season s1. Taking screenshot.")
        page.screenshot(path="verification/timeout.png")
        raise e

    # 4. Check for Play Game button
    play_btn = page.get_by_role("button", name="Play Game")
    if play_btn.is_visible():
        print("Play Game button found!")
        play_btn.click()

        # 5. Check if game container appears
        page.wait_for_selector("#game-sim-container", timeout=10000)
        print("Game container loaded!")

        # Take screenshot of the game
        page.screenshot(path="verification/live_game.png")
    else:
        print("Play Game button NOT found.")
        page.screenshot(path="verification/dashboard_no_button.png")

if __name__ == "__main__":
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        try:
            verify_live_game(page)
        except Exception as e:
            print(f"Error: {e}")
        finally:
            browser.close()
